testing/screenshots/index.js

```diff
@@ -46,25 +47,53 @@ function getOptions() {
 }
 
 (async function(){
+    process.argv.forEach((val, index) => {
+        console.log(`${index}: ${val}`);
+      });
+
     const {files, pullRequest} = getOptions();
     console.log("Taking screenshots of", files);
 
     const browser = await puppeteer.launch({ headless: true });
     const screenshots = [];
 
+    var additionalQueryArgs = process.env.ADDITIONAL_QUERY_ARGS
+    if ( additionalQueryArgs.length > 0 && !additionalQueryArgs.startsWith(`?`) ) {
+        additionalQueryArgs = `?` + additionalQueryArgs
+    }
+
     await Promise.all(files.map(async file => {
         const absolute = path.resolve(__dirname, "../..", file);
         const base = path.basename(file).replace(".json", "");
         const definition = require(absolute);
         const uid = definition.uid;
 
-        const page = await browser.newPage();
-        await page.setViewport({ width: 1440, height: 1080 });
-        await page.goto(`http://localhost:8080/grafana/d/${uid}`);
-        await page.waitForNetworkIdle();
-        const ssPath = path.join(__dirname, `${base}.png`);
-        await page.screenshot({ path: ssPath });
-        screenshots.push(ssPath);
+        if (uid == null) {
+            console.error("Missing UID for", absolute)
+            return
+        }
+
+        // Recommended resolution is 1920x1080
```

Comment:
This seems like a good candidate for refactoring if we want to add more resolutions, though I'm happy as it is for now - more of a "later down the line" thing.

Reply:
Yeah I looked at starting an array of `page` to then just iterate over. It started to add a bit of complexity whilst I was still trying to solve the delay issue so left it out for now.